db/controller/controller.js

```
@@ -45,19 +42,45 @@ exports.getArticles = (req, res) => {
 };
 
 exports.getArticleIdComments = (req, res, next) => {
-  const {article_id} = req.params
+  const { article_id } = req.params;
 
-  const promises = [fetchArticleIdComments(article_id)]
+  const promises = [fetchArticleIdComments(article_id)];
   if (article_id) {
+    promises.push(checkExists("articles", "article_id", article_id));
+  }
+
+  Promise.all(promises)
+    .then((resolvedPromises) => {
+      const commentsOnArticles = resolvedPromises[0];
+      res.status(200).send({ commentsOnArticles });
+    })
+    .catch(next);
+};
+
+
+
+
+exports.postArticleIdComments = (req, res, next) => {
+  
+  const {article_id} = req.params
+  const newComment = req.body
+
+  const promises = [insertArticleComment(newComment, article_id)]
+
+  if (newComment) {
+    promises.push(checkExists("users", "username", newComment.user))
```

If a username doesn't exist it should throw a psql error in the model which we would then handle inside our psql error handler.